static/js/modules/typeahead.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+var Handlebars = require('handlebars');
+var events = require('./events.js');
+
+module.exports = {
+  init: function() {
+    function filterCandidates(result) {
```

Comment:
I think it would be nice if we could move these functions defined within the `init()` method outside of the `module.exports` object. You will still be able to call them from within `init()` and then `init()` reads more clearly.